source.go

```
@@ -79,11 +80,16 @@ func (c SourceConfig) AWSLoadOpts() []func(*config.LoadOptions) error {
 	}
 
 	if c.AWSAssumeRoleArn != "" {
-		opts = append(opts,
-			config.WithAssumeRoleCredentialOptions(func(o *stscreds.AssumeRoleOptions) {
-				o.RoleARN = c.AWSAssumeRoleArn
-			}),
-		)
+		fmt.Println("Use assume role")
+		// Load default AWS config with our options
+		cfg, err := config.LoadDefaultConfig(ctx, opts...)
+		if err != nil {
+			fmt.Printf("Error loading AWS config: %v\n", err)
```

Can we return this error instead of ignoring it?